### PR TITLE
[HOPSWORKS-2367] Storage Connectors docs

### PR DESCRIPTION
**Pull request by jimdowling**
https://logicalclocks.atlassian.net/browse/HOPSWORKS-2367

### CODE REVIEW

File: docs/integrations/storage-connectors/snowflake.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+* Cluster identifier: The name of the cluster
+* Database driver: You can use the default JDBC Snowflake Driver `com.snowflake.jdbc42.Driver` (More on this later)
+* Database endpoint: The endpoint for the database. Should be in the format of `[UUID].eu-west-1.snowflake.com`
+* Database name: The name of the database to query
+* Database port: The port of the cluster. Defaults to 5349
+
+
+There are two options available for authentication. The first option is to configure a username and a password. The second option is to configure an IAM role. With IAM roles, Jobs or notebooks launched on Hopsworks do not need to explicitly authenticate with Snowflake, as the HSFS library will transparently use the IAM role to acquire a temporary credential to authenticate the specified user.
+
+With regards to the database driver, the library to interact with Snowflake *is not* included in Hopsworks - you need to upload the driver yourself. First, you need to [download the library](https://). You then upload the driver files to the “Resources” dataset in your project, see the screenshot below.
```
Comment (ErmiasG) [inline docs/integrations/storage-connectors/snowflake.md:22]: [download the jdbc driver] (https://repo1.maven.org/maven2/net/snowflake/snowflake-jdbc/) and [ the snowflake spark connector] (https://docs.snowflake.com/en/user-guide/spark-connector-install.html).

File: docs/integrations/storage-connectors/snowflake.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    <figcaption>When you start a Jupyter notebook, you need to add the driver so it can be accessed in programs.</figcaption>
+  </figure>
+</p>
+
+Then, you add the file to your notebook or job before launching it, as shown in the screenshots below.
+
+<p align="center">
+  <figure>
+    <img src="../../../assets/images/storage-connectors/snowflake-upload-driver.png" alt="Upload the Snowflake the driver.">
+    <figcaption>Upload the Snowflake the driver to Hopsworks.</figcaption>
```
Comment (ErmiasG) [inline docs/integrations/storage-connectors/snowflake.md:36]: Upload the JDBC driver and Snowflake Spark connector to Hopsworks.

**Review by ErmiasG — CHANGES_REQUESTED** [review]

File: docs/integrations/storage-connectors/snowflake.md (hunk 1 above)
Comment (jimdowling) [reply docs/integrations/storage-connectors/snowflake.md]: Done

File: docs/integrations/storage-connectors/snowflake.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+In the UI for the Snowflake connector, you should enter the following:
+
+* Cluster identifier: The name of the cluster
+* Database driver: You can use the default JDBC Snowflake Driver `com.snowflake.jdbc42.Driver` (More on this later)
+* Database endpoint: The endpoint for the database. Should be in the format of `[UUID].eu-west-1.snowflake.com`
+* Database name: The name of the database to query
+* Database port: The port of the cluster. Defaults to 5349
+
+
+There are two options available for authentication. The first option is to configure a username and a password. The second option is to configure an IAM role. With IAM roles, Jobs or notebooks launched on Hopsworks do not need to explicitly authenticate with Snowflake, as the HSFS library will transparently use the IAM role to acquire a temporary credential to authenticate the specified user.
```
[the remark this answers is not in the document]
Comment (jimdowling) [reply docs/integrations/storage-connectors/snowflake.md]: Done

File: docs/integrations/storage-connectors/snowflake.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  <figure>
+    <img src="../../../assets/images/storage-connectors/snowflake.png" alt="Setup a Snowflake storage connector">
+    <figcaption>Configure the Snowflake storage connector in the Hopsworks UI.</figcaption>    
+  </figure>
+</p>
+
+In the UI for the Snowflake connector, you should enter the following:
+
+* Cluster identifier: The name of the cluster
+* Database driver: You can use the default JDBC Snowflake Driver `com.snowflake.jdbc42.Driver` (More on this later)
```
[the remark this answers is not in the document]
Comment (jimdowling) [reply docs/integrations/storage-connectors/snowflake.md]: Done

File: docs/integrations/storage-connectors/redshift.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  </figure>
+</p>
+
+Then, you add the file to your notebook or job before launching it, as shown in the screenshots below.
+
+
+<p align="center">
+  <figure>
+    <img src="../../../assets/images/storage-connectors/redshift-upload-driver.png" alt="Upload the Redshift the driver.">
+    <figcaption>Upload the Redshift the driver to Hopsworks.</figcaption>
```
[the remark this answers is not in the document]
Comment (jimdowling) [reply docs/integrations/storage-connectors/redshift.md]: Done

**Comment by jimdowling** [comment]
I think i have fixed the comments.
I managed to test the building and serving of the docs. 
We are missing a snowflake UI until it is done for the new UI.

**Review by ErmiasG — APPROVED** [review]

**Comment by moritzmeister** [comment]
@jimdowling can you rebase this, then I will merge it :)

**Comment by jimdowling** [comment]
Rebased, updated Azure image to new UI as well.

File: docs/integrations/storage-connectors/snowflake.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+- role: the default security role to use for the session after connecting.
+- table: the table to which data is written to or read from. 
+
+Additional snowflake options can be added as a list of key-value pair in sfOptions
+
+There are two options available for authentication. The first option is to configure a username and a password. The second option is to use an OAuth token. See [Configure Snowflake OAuth](https://docs.snowflake.com/en/user-guide/oauth-custom.html) for instruction on how to configure OAuth support for snowflake, and [Using External OAuth](https://docs.snowflake.com/en/user-guide/spark-connector-use.html#using-external-oauth) on how you can use External OAuth to authenticate to Snowflake.
+
+With regards to the database driver, the library to interact with Snowflake *is not* included in Hopsworks - you need to upload the driver yourself. First, you need to [download the jdbc driver](https://repo1.maven.org/maven2/net/snowflake/snowflake-jdbc) and to use snowflake as the data source in spark [the snowflake spark connector](https://docs.snowflake.com/en/user-guide/spark-connector-install.html).
```
Comment (moritzmeister) [inline docs/integrations/storage-connectors/snowflake.md:31]: This page is not properly formatted. something is wrong with the bullet points.

image

**Review by moritzmeister — CHANGES_REQUESTED** [review]
The snowflake page needs some fixing for the layout.

File: docs/integrations/storage-connectors/snowflake.md (hunk 6 above)
Comment (jimdowling) [reply docs/integrations/storage-connectors/snowflake.md]: I fixed the bullet points. When i run 'mkdocs serve', it looks ok.

**Review by moritzmeister — APPROVED** [review]

**Pull request merged**